es-app/src/components/TextListComponent.h

```diff
@@ -76,13 +77,11 @@ class TextListComponent : public IList<TextListData, T>
 	inline void setSelectorOffsetY(float selectorOffsetY) { mSelectorOffsetY = selectorOffsetY; }
 	inline void setSelectorColor(unsigned int color) { mSelectorColor = color; }
 	inline void setSelectedColor(unsigned int color) { mSelectedColor = color; }
-	inline void setScrollSound(const std::shared_ptr<Sound>& sound) { mScrollSound = sound; }
 	inline void setColor(unsigned int id, unsigned int color) { mColors[id] = color; }
-	inline void setSound(const std::shared_ptr<Sound>& sound) { mScrollSound = sound; }
 	inline void setLineSpacing(float lineSpacing) { mLineSpacing = lineSpacing; }
 
 protected:
-	virtual void onScroll(int amt) { if(mScrollSound) mScrollSound->play(); }
+	virtual void onScroll(int amt) { if(!mScrollSound.empty() && Settings::getInstance()->getBool("EnableSounds")) { Sound::get(mScrollSound)->play(); } }
```

Comment:
As an alternative to having to check the setting everywhere that we may want to play a sound.  Could we have something along the lines of `Sound::play(mScrollSound)` that handles it?

Reply:
There is already a check here: https://github.com/RetroPie/EmulationStation/blob/master/es-core/src/Sound.cpp#L114

However, Sound::get() itself will trigger AudioManager::getInstance(), which will open the audio device *before* reaching the Sound::play() check (even if EnableSounds is false). Adding the additional check here avoids unnecessarily opening the audio device in this circumstance.

Reply:
I understand that, but ideally the TextListComponent shouldn't even be aware that there is a setting to enable/disable sounds.    To be more clear, I am suggesting adding something along the lines of the following to Sound.cpp: 

```cpp
void Sound::play(const std::string& path)
{
     if(!mScrollSound.empty() && Settings::getInstance()->getBool("EnableSounds")) 
    { 
        Sound::get(mScrollSound)->play(); 
    }
}
```

Also, shouldn't these need to be adjusted as well?

https://github.com/RetroPie/EmulationStation/blob/master/es-app/src/views/gamelist/IGameListView.cpp#L16

https://github.com/RetroPie/EmulationStation/blob/master/es-app/src/views/gamelist/ISimpleGameListView.cpp

Reply:
Thanks for the feedback - I've updated the patchset.

TextListComponent no longer checks the bool. Instead a check is added to AudioManager::getInstance(), which means that it will also catch the extra cases you identified.